pkg/helmreconciler/rendering.go

```diff
@@ -240,7 +240,7 @@ func (h *HelmReconciler) ProcessObject(obj *unstructured.Unstructured) error {
 				log.Errorf("unexpected error occurred during postprocessing of updated resource: %s", err)
 			}
 		} else {
-			listenerErr := h.customizer.Listener().ResourceError(mutatedObj, err)
+			listenerErr := h.customizer.Listener().ResourceError(obj, err)
```

Comment:
Thanks for looking into this Iris. I find this function (and especially this block) quite difficult to read and I think it could be structured better to avoid having so much nesting. I wouldn't be surprised if we find more errors of this type in here without some cleanup.